openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -761,12 +782,74 @@ def parse_smirnoff_from_source(self, source):
             input_format
             for input_format in self._parameter_io_handlers.keys()
         ]
-        msg = f"Source {source} does not appear to be in a known SMIRNOFF encoding.\n"
+        msg = f"Source {source} could not be read. If this is a file, ensure that the path is correct.\n"
+        msg += "If the file is present, ensure it is in a known SMIRNOFF encoding.\n"
         msg += f"Valid formats are: {valid_formats}\n"
-        msg += f"Parsing vailed with the following error:\n{exception_msg}\n"
+        msg += f"Parsing failed with the following error:\n{exception_msg}\n"
         raise IOError(msg)
 
 
+    def to_string(self, format='XML', discard_cosmetic_attributes=True):
+        """
+        Write this Forcefield and all its associated parameters to a string in a given format which
+        complies with the SMIRNOFF spec.
+
+
+        Parameters
+        ----------
+        format : str, optional. Default='XML'
```

Comment:
We could think of having formats codification here, for `io_format`, and in `toolkits.py` in an `Enum` object instead of strings (probably for a later PR/API-breaking version).

We could also extend this to accept directly a `ParameterIOHandler` object so that we could easily implement translation between specification in the future. So maybe we can call this something like `parameter_io_handler`, which accept also a parameter handler key (i.e. a format) and the code could be something like
```python
if isinstance(parameter_io_handler, ParameterIOHandler):
    io_handler = parameter_io_handler
else:
    io_handler = self. get_parameter_io_handler(parameter_io_handler)
```

Reply:
Brilliant! Agree that IO format enums should come later, but will extend this to accept `ParameterIOHandlers` right now